src/jit/importer.cpp

```diff
@@ -12377,13 +12377,10 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 {
                     if ((prefixFlags & PREFIX_TAILCALL_EXPLICIT) != 0)
                     {
-#ifdef DEBUG
-                        if (verbose)
-                        {
-                            printf("\n\nIgnoring the tail call prefix in the inlinee %s\n", info.compFullName);
-                        }
-#endif
-                        prefixFlags &= ~PREFIX_TAILCALL_EXPLICIT;
+                        // TODO: We can inline some callees with explicit tail calls if we can guarantee that the calls
+                        // can be dispatched as tail calls from the caller.
+                        compInlineResult->NoteFatal(InlineObservation::CALLEE_EXPLICIT_TAIL_PREFIX);
```

Comment:
It would seem that this observation could have been done earlier, in `fgMakeBasicBlocks` where there's already some tailcall + inlining stuff.